cmake/kokkos_functions.cmake

```diff
@@ -954,6 +954,107 @@ int main()
   set(${_VAR} ${_RET} CACHE STRING "CXX compiler supports building CUDA")
 endfunction()
 
+# this function checks if the compiler supports specific flags. No-op for CMake < 3.19:
+#
+#       SILENT    --> check without printing and erroring out (default=OFF)
+#       LANGUAGE  --> compile language (required)
+#       FLAGS     --> list of compiler flags
+#       OPTIONS   --> list of options that get printed in case of failure (optional)
+#       RESULT    --> bool with test result (output, optional)
+#
+function(kokkos_check_compiler_flags)
+  cmake_parse_arguments(INP "SILENT" "LANGUAGE;RESULT" "FLAGS;OPTIONS" ${ARGN})
+
+  if(NOT INP_LANGUAGE)
+    message(FATAL_ERROR "'kokkos_check_compiler_flags' requires a LANGUAGE to be passed.")
+  endif()
+
+  #if no check is performed return true
+  if(DEFINED INP_RESULT)
+    set(${INP_RESULT} ON PARENT_SCOPE)
+  endif()
+
+  set(CMAKE_REQUIRED_QUIET ON)
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.19)
+    include(CheckCompilerFlag)
+    if(INP_FLAGS)
+      #check_compiler_flag requires a whitespace separated list
+      string(REPLACE ";" " " WHITESPACE_FLAGS "${INP_FLAGS}")
+      # sycl adds "-device ..." options that need quotes (which CMake removes). We need to add them here again.
+      string(REGEX REPLACE "(-device [A-Za-z0-9_\\\\.]*)" "\"\\1\"" QUOTED_FLAGS ${WHITESPACE_FLAGS})
```

Comment:
@masterleinad was right, sycl requires the argument `-device 12....` to be in quotes for the check to pass. In lack of a better strategy I just re-add the quotation marks here.

Open to suggestions how to improve that.

Reply:
We should instead quote the compiler flag like
```diff
diff --git a/cmake/kokkos_arch.cmake b/cmake/kokkos_arch.cmake
index 1e3e8cb8a..b64f80b90 100644
--- a/cmake/kokkos_arch.cmake
+++ b/cmake/kokkos_arch.cmake
@@ -1081,9 +1081,15 @@ if(KOKKOS_ENABLE_SYCL)
     elseif(KOKKOS_ARCH_INTEL_XEHP)
       set(SYCL_TARGET_BACKEND_FLAG -Xsycl-target-backend "-device 12.50.4")
     elseif(KOKKOS_ARCH_INTEL_PVC)
-      set(SYCL_TARGET_BACKEND_FLAG -Xsycl-target-backend "-device 12.60.7")
+      set(SYCL_TARGET_BACKEND_FLAG "-Xsycl-target-backend \"-device 12.60.7\"")
     endif()
```

Reply:
https://github.com/kokkos/kokkos/pull/7928

Reply:
that definitely is the better solution

Reply:
we tried: https://github.com/kokkos/kokkos/pull/7928 TLDR: did not work and we don't know why ... it results in `ycl-target-backend "-device 12.60.7"` on the compile line ...

Reply:
I investigated a bit: CMake seems to properly propagate "-Xsycl-target-backend \"-device 12.60.7\"" to the compile line, then icpx reads this as -ycl-target-backend "-device 12.60.7" ...

Reply:
Yes, I have no idea how `icpx` parses the string to pass a truncated version to `ocloc`.